keras_cv/layers/regularization/dropblock_2d.py

```diff
@@ -0,0 +1,191 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+from keras.engine.base_layer import BaseRandomLayer
+from keras.utils import conv_utils
+
+
+class DropBlock2D(BaseRandomLayer):
+    """Applies DropBlock regularization to input features.
+
+    DropBlock is a form of structured dropout, where units in a contiguous
+    region of a feature map are dropped together. DropBlock works better than
+    dropout on convolutional layers due to the fact that activation units in
+    convolutional layers are spatially correlated.
+
+    It is advised to use DropBlock after activation in Conv -> BatchNorm -> Activation
+    block in further layers of the network. For example, the paper mentions using
+    DropBlock in 3rd and 4th group of ResNet blocks.
+
+    Reference:
+    - [DropBlock: A regularization method for convolutional networks](
+        https://arxiv.org/abs/1810.12890
+    )
+
+    Args:
+        keep_probability: float. Probability of keeping a unit. Defaults to 0.9.
+            Must be between 0 and 1. For best results, the value should be between
+            0.75-0.95
+        dropblock_size: integer. The size of the block to be dropped. Defaults to 7.
+            Must be bigger than 0, and should not be bigger than the input feature map
+            size. If this value is greater by 1 from the input feature map size you will
+            encounter `ZeroDivisionError`.
+        data_format: string. One of channels_last (default) or channels_first. The
+            ordering of the dimensions in the inputs. channels_last corresponds to
+            inputs with shape (batch_size, height, width, channels) while channels_first
+            corresponds to inputs with shape (batch_size, channels,height, width). It
+            defaults to the image_data_format value found in your Keras config file at
+            ~/.keras/keras.json. If you never set it, then it will be channels_last.
+        seed: integer. To use as random seed.
+        name: string. The name of the layer.
+
+    Usage:
+        DropBlock2D should be used inside a `tf.keras.Model`:
```

Comment:
let's actually change this to DropBlock2D *can* be used inside a `tf.keras.Model`

We don't want to make assumptions about usage patterns users will follow

Reply:
Changed should -> can.